[torchgen] Extract out schema registration logic into a function

larryliu0820 opened the pull request:
Summary:
A followup to  #78015 and #79733. In those PRs I introduced custom namespace support into:
* `Register<DispatchKey>.cpp`
* `RegisterSchema.cpp`
* `NativeFunctions.h`

This PR extracts out logic that generates schema registration code (used in `RegisterSchema.cpp`) into a function so that it can be easily tested and reused. Added unit test to cover the logic as well.

Test Plan: Rely on newly added unit tests.

Differential Revision: D37581186

iseeyuan reviewed (approved):
Nice refactor and test coverage!

larryliu0820 commented:
@pytorchbot merge

larryliu0820 committed:
[torchgen] Extract out schema registration logic into a function (#80780)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/80780

A followup to  #78015 and #79733. In those PRs I introduced custom namespace support into:
* `Register<DispatchKey>.cpp`
* `RegisterSchema.cpp`
* `NativeFunctions.h`

This PR extracts out logic that generates schema registration code (used in `RegisterSchema.cpp`) into a function so that it can be easily tested and reused. Added unit test to cover the logic as well.

Test Plan: Rely on newly added unit tests.

Reviewed By: bdhirsh

Differential Revision: D37581186

fbshipit-source-id: 01ca31e9293bc17ae2c6445995399a44cc4aa737

larryliu0820 commented:
@pytorchbot merge
